pageserver/src/tenant/timeline.rs

```diff
@@ -2143,13 +2143,31 @@ impl Timeline {
     ///
     /// This function takes the current timeline's locked LayerMap as an argument,
     /// so callers can avoid potential race conditions.
+    ///
+    // TODO: find a way to not hold the Timeline::layers lock during get_value_reconstruct_data calls.
+    //
+    // Since these calls do local disk IO, they'll be reasonably fast, until come disk IOPS bound.
+    // We have lots of headroom on current pageservers, so, it's going to be fine for now.
+    //
+    // We can't use tokio::sync::RwLock that easily because its guard is not Send, but,
+    // many tasks that access Timeline::layers run inside task_mgr tasks, which are required
+    // to be Send. It has been tried in origin/problame/asyncify-get-reconstruct-data--tokio-sync.
```

Comment:
https://docs.rs/tokio/latest/tokio/sync/struct.OwnedRwLockReadGuard.html#impl-Send-for-OwnedRwLockReadGuard%3CT,+U%3E I don't see why it would not be sendable?

Reply:
That is indeed confounding, will spend some more time figuring out why my other branch where I tried to use tokio::sync instead of parking_lot miscompiled.